### PR TITLE
MAT-7959: fix validation for Code field in Codes tab and Definition N…

### PR DESCRIPTION
**Pull request by etan-sb**
…ame field in Definitions tab

## MADiE PR

Jira Ticket: [MAT-7959](https://jira.cms.gov/browse/MAT-7959)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is in to the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/CqlBuilderPanel/codesSection/codesSubSection/codeSubSection/CodeSection.tsx [hunk 1]
```diff
@@ -127,8 +128,9 @@ export default function CodeSection({
     ),
   };
 
-  function trimCodeInput() {
-    formik.setFieldValue("code", formik.values.code.trim());
+  function trimCodeInput(values) {
+    values.code = values.code.trim();
+    formik.setFieldValue("code", values.code);
```
Comment (sb-prateekkeerthi) [inline src/CqlBuilderPanel/codesSection/codesSubSection/codeSubSection/CodeSection.tsx]: can we remove the line 132 and directly write 133 like: formik.setFieldValue("code", values.code.trim());

can you try if this working

File: src/CqlBuilderPanel/codesSection/codesSubSection/codeSubSection/CodeSection.tsx [hunk 2]
```diff
@@ -74,6 +74,7 @@ export default function CodeSection({
     validationSchema: CodeSubSectionSchemaValidator,
     enableReinitialize: true,
     onSubmit: (values) => {
+      trimCodeInput(values);
```
Comment (sb-prateekkeerthi) [inline src/CqlBuilderPanel/codesSection/codesSubSection/codeSubSection/CodeSection.tsx]: can you write a comment here saying that when validations are being added onBlur trim code is not working as expected so we had to manually call that again

Comment (etan-sb) [reply src/CqlBuilderPanel/codesSection/codesSubSection/codeSubSection/CodeSection.tsx]: I was able to keep the onBlur functionality while checking for validation on the code field by adding the line `formik.setFieldTouched("code", true);` in the onBlur handler

**Review by sb-prateekkeerthi — APPROVED** [review]
